src/modules/mc_pos_control/Takeoff/Takeoff.cpp

```diff
@@ -38,73 +38,119 @@
+		: _parent(parent),
+		  _armed(armed), 
+		  _landed(landed), 
+		  _want_takeoff(want_takeoff), 
+		  _skip_takeoff(skip_takeoff), 
+		  _takeoff_desired_vz(takeoff_desired_vz), 
+		  _now_us(now_us)
+	{
+	}
 
-	// FALLTHROUGH
-	case TakeoffState::spoolup:
-		if (_spoolup_time_hysteresis.get_state()) {
-			_takeoff_state = TakeoffState::ready_for_takeoff;
+	Takeoff *_parent;
+	bool _armed, _landed, _want_takeoff, _skip_takeoff;
+	float _takeoff_desired_vz;
+	const hrt_abstime &_now_us;
+};
 
-		} else {
-			break;
-		}
+class Takeoff::StateHandler
+{
+public:
+	static void handleDisarmed(const StateParams& params);
+	static void handleSpoolUp(const StateParams& params);
+	static void handleReadyForTakeoff(const StateParams& params);
+	static void handleRampUp(const StateParams& params);
+	static void handleFlight(const StateParams& params);
+};
+
+void Takeoff::StateHandler::handleDisarmed(const StateParams& params)
+{
+	if (params._armed)
+	{
+		params._parent->_takeoff_state = TakeoffState::spoolup;
+		handleSpoolUp(params);
+	}
+}
 
-	// FALLTHROUGH
-	case TakeoffState::ready_for_takeoff:
-		if (want_takeoff) {
-			_takeoff_state = TakeoffState::rampup;
-			_takeoff_ramp_vz = _takeoff_ramp_vz_init;
+void Takeoff::StateHandler::handleSpoolUp(const StateParams& params)
+{
+	if (params._parent->_spoolup_time_hysteresis.get_state()) 
+	{
+		params._parent->_takeoff_state = TakeoffState::ready_for_takeoff;
+		handleReadyForTakeoff(params);
+	}
+}
 
-		} else {
-			break;
-		}
+void Takeoff::StateHandler::handleReadyForTakeoff(const StateParams& params)
+{
+	if (params._want_takeoff) {
+		params._parent->_takeoff_state = TakeoffState::rampup;
+		params._parent->_takeoff_ramp_vz = params._parent->_takeoff_ramp_vz_init;
+		handleRampUp(params);
+	}
+}
 
-	// FALLTHROUGH
-	case TakeoffState::rampup:
-		if (_takeoff_ramp_vz >= takeoff_desired_vz) {
-			_takeoff_state = TakeoffState::flight;
+void Takeoff::StateHandler::handleRampUp(const StateParams& params)
+{
+	if (params._parent->_takeoff_ramp_vz >= params._takeoff_desired_vz) 
+	{
+		params._parent->_takeoff_state = TakeoffState::flight;
+		handleFlight(params);
+	}
+}
 
-		} else {
-			break;
-		}
+void Takeoff::StateHandler::handleFlight(const StateParams& params)
+{
+	if (params._landed) {
+		params._parent->_takeoff_state = TakeoffState::ready_for_takeoff;
+	}
+}
 
-	// FALLTHROUGH
-	case TakeoffState::flight:
-		if (landed) {
-			_takeoff_state = TakeoffState::ready_for_takeoff;
-		}
+void Takeoff::generateInitialRampValue(const float hover_thrust, float velocity_p_gain)
+{
+	velocity_p_gain = math::max(velocity_p_gain, 0.01f);
+	_takeoff_ramp_vz_init = -hover_thrust / velocity_p_gain;
+}
 
-		break;
+void Takeoff::updateTakeoffState(const bool armed, const bool landed, const bool want_takeoff,
+				 const float takeoff_desired_vz, const bool skip_takeoff, const hrt_abstime &now_us)
+{
+	_spoolup_time_hysteresis.set_state_and_update(armed, now_us);
 
-	default:
-		break;
+	if (!armed)	{
```

Comment:
spacing

Reply:
apply clang-tidy format to file

Reply:
We use astyle. To format your code locally before uploading just run `make format` and it will automatically be corrected.